docs/counter_example.md

```diff
@@ -1,53 +1,102 @@
-# ABCI Counter with AEAs
+# Counter Agent Service Demo
 
-This demo shows how to run an ABCI counter application
-over Tendermint.
-It is an example on how to replicate an application state-machine
-across different AEAs using the Tendermint consensus engine.
+This demo shows how to run an agent service implementing a counter.
+It is simple example that illustrates how state-machine replication is achieved across different agents through the consensus gadget, which is Tendermint, in this case. Note that, unlike the remaining demos, due to its simplicity, the business logic of this service is not encoded as an {{fsm_app}}. Rather, it is implemented as a simple skill (`counter`) containing an `ABCIHandler`. The skill does not contain proactive behaviours, which means that no client calls are made to the consensus gadget.
 
-## Preliminaries
+## Architecture of the Demo
 
-Make sure you have installed on your machine:
+This demo is composed of:
 
-- [Docker Engine](https://docs.docker.com/engine/install/)
-- [Docker Compose](https://docs.docker.com/compose/install/)
+- A set of four [Tendermint](https://tendermint.com/) nodes (`node0`, `node1`, `node2`, `node3`).
+- A set of four AEAs (`abci0`, `abci1`, `abci2`, `abci3`), in one-to-one connection with their corresponding Tendermint
+node.
 
-## Run
+<figure markdown>
+![](./images/counter_diagram.svg)
+<figcaption>Architecture of the Counter demo</figcaption>
+</figure>
 
-To set up the network:
+{% include 'requirements.md' %}
 
-```
-make localnet-start
-```
 
-The network is made of:
+## Running the Demo
+
+The steps below will guide you to create a Pipenv environment for the demo,
+download the counter agent service definition from the Service Registry
+and build a deployment that will run locally.
+
+1. Open a terminal and create a workspace folder, e.g.,
+    ```bash
+    mkdir my_demo
+    cd my_demo
+    ```
+
+2. Within the workspace folder, setup the environment:
+    ```bash
+    export VERSION=0.1.0
+    export OPEN_AEA_IPFS_ADDR="/dns/registry.autonolas.tech/tcp/443/https"
+    touch Pipfile && pipenv --python 3.10 && pipenv shell
```

Comment:
Have you run this? Make sure that they follow the quickstart steps. I think there is a missing autonomy init needed for those that never had run anything before.

Reply:
We should assume that the setup has happened in the quickstart. So the demos should not contain these setup steps at all. Just say, make sure you're set up...

Reply:
Fixed.